axelrod/game.py

```diff
@@ -110,10 +139,19 @@ def __init__(self, r: Score = 3, s: Score = 0, t: Score = 5, p: Score = 1) -> No
             Score obtained by a player for defecting against a cooperator.
         p: int or float
             Score obtained by both player for mutual defection.
+        **attributes
+            optional attributes for the game. Used
+            to ensure strategies used with the game are valid.
+
         """
         A = np.array([[r, s], [t, p]])
 
-        super().__init__(A, A.transpose())
+        default_characteristics = {
+            'has_RPST': hasattr(self, 'RPST')
```

Comment:
We don't really need a distinguished `has_RPST` characteristic since R, P, S, and T are just the values of the game matrix and so any 2x2 matrix game provides them. For clarity in the calling strategies I think we should still have an RPST function though. The difference is that the strategies need the values to be well-defined but the game itself always has them, so the requirement is really "game can provide RPST values", which wouldn't be possible for say non-linear payoffs.

Reply:
fine by me - in that case I'd suggest refactoring RPST in a different PR, and I'll remove the default characteristic for now.